### PR TITLE
Tensorflow for digits

### PR DESCRIPTION
**Pull request by ethantang95**
I corrected the commit squashing, should be more organized

@TimZaman, @gheinrich, @lukeyeager

### CODE REVIEW

**Comment by TimZaman** [comment]
Ah that looks more sane

File: digits/dataset/images/classification/forms.py [hunk 1]
```diff
@@ -21,13 +21,16 @@ class ImageClassificationDatasetForm(ImageDatasetForm):
                                   choices=[
                                       ('lmdb', 'LMDB'),
                                       ('hdf5', 'HDF5'),
+                                      ('tfrecords', 'TFRecords'),
```
Comment (ethantang95) [inline digits/dataset/images/classification/forms.py]: So we do not support TF record anymore, do we keep it or remove it?

File: digits/dataset/images/classification/forms.py [hunk 2]
```diff
@@ -21,13 +21,16 @@ class ImageClassificationDatasetForm(ImageDatasetForm):
...
                                   ],
                                   default='lmdb',
                                   )
 
     def validate_backend(form, field):
         if field.data == 'lmdb':
             form.compression.data = 'none'
+        elif field.data == 'tfrecords':
+            form.compression.data = 'none'
```
Comment (ethantang95) [inline digits/dataset/images/classification/forms.py]: See comment above

File: digits/dataset/tasks/create_db.py [hunk 3]
```diff
@@ -50,8 +50,9 @@ def __init__(self, input_file, db_name, backend, image_dims, **kwargs):
         self.input_file = input_file
         self.db_name = db_name
         self.backend = backend
-        if backend == 'hdf5':
+        if backend == 'hdf5' or backend == 'tfrecords':
```
Comment (ethantang95) [inline digits/dataset/tasks/create_db.py]: See comment above

**Comment by TimZaman** [comment]
Keep, in case we want to enable it in the future, f.e. When people want
more perf

File: digits/model/images/generic/test_views.py [hunk 4]
```diff
@@ -1255,3 +1308,55 @@ class TestAllInOneNetwork(BaseTestCreation, BaseTestCreated, test_utils.CaffeMix
...
+#
+# class TestTensorflowCreatedWithImageProcessingExtensionMeanPixel(BaseTestCreatedWithImageProcessingExtension,
+#                                                                  test_utils.TensorflowMixin):
+#     MEAN = 'pixel'
+#
+#
+# class TestTensorflowCreatedWithImageProcessingExtensionMeanNone(BaseTestCreatedWithImageProcessingExtension,
+#                                                                 test_utils.TensorflowMixin):
+#     MEAN = 'none'
+
```
Comment (ethantang95) [inline digits/model/images/generic/test_views.py]: What tests were these suppose to be for?

File: digits/standard-networks/tensorflow/gradient.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
+
```
Comment (ethantang95) [inline digits/standard-networks/tensorflow/gradient.py]: Remove this?

File: digits/standard-networks/tensorflow/lenet_slim.py [hunk 6]
```diff
@@ -0,0 +1,27 @@
+class UserModel(Tower):
```
Comment (ethantang95) [inline digits/standard-networks/tensorflow/lenet_slim.py]: We have both standard and slim versions... but they are both buried under standard-networks. Should the slim be move somewhere else for more visibility? the default one will be the standard one?

File: digits/standard-networks/tensorflow/vgg16.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
+class UserModel(Tower):
```
Comment (ethantang95) [inline digits/standard-networks/tensorflow/vgg16.py]: What should we do with this?

**Pull request closed without merging**